### ISSUE TITLE
WC_Coupon set_used_by doesn't work, must use increase_usage_count and decrease_usage_count for it

### ISSUE THREAD

**Issue by senadir** [woothemes/woocommerce#38214]
`set_used_by` seems to be ignored, doing things like

```php
$coupon->set_used_by( array( 45, 50, 60 ) );
$coupon->save();
```
Doesn't persist the value at all, even `set_used_by( 45 )` or `set_used_by( '45' )`.

Calling `get_used_by` returns an array of strings instead of numbers.

To actually set that field value, you need to call `increase_usage_count( '45' )` and to remove a value you need to call `decrease_usage_count( '45' )` which seems wrong since I expect that to only increase usage count.

**Comment by barryhughes** [comment]
To add, there is a possible hint as to what's going wrong [in this comment](https://github.com/woocommerce/woocommerce/blob/7.7.0/plugins/woocommerce/includes/class-wc-coupon.php#L818), but it does seem that we should either fix this so the code shared by @senadir works as expected or, at the very least, clearly document (in the docblock for `WC_Coupon::set_used_by()`) why it does not work.

**Comment by pmcpinto** [comment]
This needs engineering input before we can confirm the final disposition.

**Comment by senadir** [comment]
Still valid thou low priority a bit
